parser_test.go

```diff
@@ -40,30 +40,64 @@ func TestParseSuccess(t *testing.T) {
 	same(`{}[:3.4s] offset -10y`)
 	same(`{Foo="bAR"}`)
 	same(`{foo="bar"}`)
+	another(`{"foo"="bar"}`, `{foo="bar"}`)
+	another(`{"foo"="bAR"}`, `{foo="bAR"}`)
+	another(`{"3foo"="bar"}`, `{\3foo="bar"}`)
```

Comment:
do you think the expected output should be `{"3foo="bar"}`? I mean use quotting instead of escaping. Quotting seems more user friendly than escaping.
Although, metricsql parser is used in QueryPrettifier (see button next to query expression field in vmui) that would apply pretty formatting to expression.